examples/js/plugins/FeatureToolTip.js

```diff
@@ -1,150 +1,224 @@
-    };
-
-    var layer;
-    var result;
-    var feature;
-    var symb;
-    var style;
-    var fill;
-    var stroke;
-    var content;
-    var prop;
-    var layers = viewer.getLayers(function _(l) { return l.source && l.source.isFileSource; });
-    function buildToolTip(geoCoord, e) {
-        var visible = false;
-        var i = 0;
-        var p = 0;
-        var precision = viewer.controls.pixelsToDegrees(opts.precisionPx || 5);
+    }, false);
 
+    function moveToolTip(event) {
         tooltip.innerHTML = '';
         tooltip.style.display = 'none';
-        if (geoCoord) {
-            visible = false;
-
-            // Pick on each layer, and display them all in the tooltip
-            for (i = 0; i < layers.length; i++) {
-                layer = layers[i];
-
-                if (!layer.source.parsedData) { continue; }
-
-                result = itowns.FeaturesUtils.filterFeaturesUnderCoordinate(geoCoord, layer.source.parsedData, precision);
-
-                for (p = 0; p < result.length; p++) {
-                    content = '';
-                    visible = true;
-
-                    feature = result[p].geometry;
-                    style = layer.style.isStyle ? layer.style : feature.properties.style;
-                    fill = style.fill.color;
-                    stroke = '1.25px ' + style.stroke.color;
-
-                    if (result[p].type === itowns.FEATURE_TYPES.POLYGON) {
-                        symb = '&#9724';
-                    } else if (result[p].type === itowns.FEATURE_TYPES.LINE) {
-                        symb = '&#9473';
-                        fill = style.stroke.color;
-                        stroke = '0px';
-                    } else if (result[p].type === itowns.FEATURE_TYPES.POINT) {
-                        symb = '&#9679';
-                    }
 
-                    content += '<div>';
-                    content += '<span style="color: ' + fill + '; -webkit-text-stroke: ' + stroke + '">';
-                    content += symb + ' ';
-                    content += '</span>';
-                    content += (feature.properties.name || feature.properties.nom || feature.properties.description || layer.name);
+        var features = view.pickFeaturesAt(event, 3, layersId);
 
-                    if (result[p].type === itowns.FEATURE_TYPES.POINT) {
-                        content += '<br/><span class="coord">long ' + result[p].coordinates[0].toFixed(4) + '</span>';
-                        content += '<br/><span class="coord">lat ' + result[p].coordinates[1].toFixed(4) + '</span>';
-                    }
+        var layer;
+        for (var layerId in features) {
+            if (features[layerId].length == 0) {
+                continue;
+            }
+
+            layer = layers[layersId.indexOf(layerId)];
+            if (typeof layer.options.filterGeometries == 'function') {
+                features[layerId] = layer.options.filterGeometries(features[layerId], layer.layer);
```

Comment:
I prefer that the user gives a function `getIdFromProperties(properties)` and that we develop with a filtering function.
May be throw a error if this function isn't exist.

Reply:
The idea here is that the user could filter the features as he wishes, not necessary by using the id.

Reply:
ok